src/Microsoft.DotNet.Build.Tasks/PackageFiles/scripts/vstsagent/CloneWithRetry.cmd

```diff
@@ -0,0 +1,35 @@
+@echo off
+
+if "%1" == ""   goto :usage
+if "%2" == ""   goto :usage
+goto :main
+
+:usage
+echo Usage: %~n0 [repository url] [destination folder]
+echo.
+echo Clones a git repo using the specified URL with depth 1 (for performance), 
+echo retrying up to 10x if there is a failure.
+goto :eof
+
+:main
+set REPO_URL=%1
+set FOLDER_NAME=%2
+set /a RETRY_COUNT=10
+set /a SLEEP_TIME=10
+
+IF NOT DEFINED GIT SET "GIT=%PROGRAMFILES(X86)%\Git\cmd\git.exe"
```

Comment:
GIT was the environment variable that the build step I was hoping to use this in was already using, the goal was to change nothing about this, but yeah could just change it to IF NOT DEFINED GIT SET GIT=git.exe and use the version on the path.   However, I'm realizing that if this is in buildtools it won't be available before the repo is cloned, brainstorming offline with chcosta now.